flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/JobAutoScalerImpl.java

```diff
@@ -102,10 +108,8 @@ public boolean scale(FlinkResourceContext<? extends AbstractFlinkResource<?, ?>>
                             resource, autoScalerInfo, ctx.getFlinkService(), conf);
 
             LOG.debug("Evaluating scaling metrics for {}", collectedMetrics);
-            var evaluatedMetrics = evaluator.evaluate(conf, collectedMetrics);
+            evaluatedMetrics = evaluator.evaluate(conf, collectedMetrics);
             LOG.debug("Scaling metrics evaluated: {}", evaluatedMetrics);
-            lastEvaluatedMetrics.put(resourceId, evaluatedMetrics);
-            flinkMetrics.registerScalingMetrics(() -> lastEvaluatedMetrics.get(resourceId));
 
             if (!collectedMetrics.isFullyCollected()) {
                 // We have done an upfront evaluation, but we are not ready for scaling.
```

Comment:
I think we should somehow null out the RECOMMENDED_PARALLELISM (or set it to Double.nan) if the metrics are not fully collected because at that point we are not actually recommending anything. It would be then consistent of not reporting the values when the job is not running / still stabilizing.

Reply:
We can do that, and I feel like null is a more safe choice the Double.nan. @mxm what do you think? Shall we keep or wipe the metrics during the metric window? Since the recommended parallelism is set to the current parallelism to me both are ok.

Reply:
My point was that you cannot simply `null` them out here because that may interfere with how the metrics are initialized.